var/spack/repos/builtin/packages/py-twisted/package.py

```diff
@@ -9,11 +9,127 @@
+    depends_on('py-automat@0.8.0:', type=('build', 'run'), when='@21.7.0:')
+    depends_on('py-hyperlink@17.1.1:', type=('build', 'run'), when='@21.7.0:')
+    depends_on('py-attrs@19.2.0:', type=('build', 'run'), when='@21.7.0:')
+    depends_on('py-typing-extensions@3.6.5:', type=('build', 'run'), when='@21.7.0:')
+
+    def url_for_version(self, version):
+        url = "https://pypi.io/packages/source/T/Twisted/"
+
+        if version <= Version('20.3.0'):
+            url += 'Twisted-{0}.tar.bz2'
+        else:
+            url += 'Twisted-{0}.tar.gz'
+
+        url = url.format(version)
+        return url
+
+    @property
+    def import_modules(self):
+        modules = ['twisted',
+                   'twisted.positioning',
+                   'twisted.positioning.test',
+                   'twisted.protocols',
+                   'twisted.protocols.test',
+                   'twisted.protocols.haproxy',
+                   'twisted.protocols.haproxy.test',
+                   'twisted.web',
+                   'twisted.web._auth',
+                   'twisted.web.test',
+                   'twisted.scripts',
+                   'twisted.scripts.test',
+                   'twisted.runner',
+                   'twisted.runner.test',
+                   'twisted.cred',
+                   'twisted.cred.test',
+                   'twisted.plugins',
+                   'twisted.enterprise',
+                   'twisted.logger',
+                   'twisted.logger.test',
+                   'twisted.persisted',
+                   'twisted.persisted.test',
+                   'twisted.names',
+                   'twisted.names.test',
+                   'twisted.pair',
+                   'twisted.pair.test',
+                   'twisted.test',
+                   'twisted.tap',
+                   'twisted.python',
+                   'twisted.python.test',
+                   'twisted.trial',
+                   'twisted.trial._dist',
+                   'twisted.trial._dist.test',
+                   'twisted.trial.test',
+                   'twisted.words',
+                   'twisted.words.protocols',
+                   'twisted.words.protocols.jabber',
+                   'twisted.words.im',
+                   'twisted.words.test',
+                   'twisted.words.xish',
+                   'twisted.spread',
+                   'twisted.spread.test',
+                   'twisted.conch',
+                   'twisted.conch.scripts',
+                   'twisted.conch.ui',
+                   'twisted.conch.client',
+                   'twisted.conch.openssh_compat',
+                   'twisted.conch.test',
+                   'twisted.conch.insults',
+                   'twisted.conch.ssh',
+                   'twisted.internet',
+                   'twisted.internet.test']
+
+        if self.spec.satisfies('@:15.4.0'):
+            modules.remove('twisted.protocols.haproxy')
```

Comment:
It would be fewer lines of code to conditionally add instead of add and then conditionally remove.

We've talked about having a function that excludes instead of includes modules. Would that help in this case? We could also support glob characters if that helps.

Reply:
Can we just ignore test failures in 15.x series (by, e.g., deprecating them)? And yes, (conditionally) blacklisting modules would help, since there is only one bad module in 21.7.0...

Reply:
We could, I don't see any packages that specifically require the 15.x series.